receipt_for_lovely_loveseats/test_checkout__add_to_checkout.py

```diff
@@ -0,0 +1,57 @@
+from unittest import TestCase
+from checkout import Checkout
+
+class TestCheckoutAddToCheckout(TestCase):
+  def setUp(self):
+    self.checkout = Checkout(products = { 'book': {}, 'chair': {}, 'desk': {}}, items = [])
```

Comment:
Experienced a weird behavior when not explicitly assigning `items` in the argument, even though the default is an empty list.

When running the test, the `items` list doesn't get cleared when the second test is run. The object from the first test persists.

This is weird. My understanding (and after playing around with the test) is that `setUp` runs before each test.

Therefore, an assignment to a newly instantiate class before a test run should reset back to the default state.

I also tried resetting `items` in a `tearDown`, but the change does not get carried over to the next test.

I may be misunderstanding how the lifecycle works...

Reply:
I believe I found the answer... Default objects that are mutable persist.

See the following:

* https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python
* https://pylint.pycqa.org/en/latest/messages/warning/dangerous-default-value.html?highlight=W0102